src/storage/v2/inmemory/storage.cpp

```diff
@@ -1210,7 +1226,7 @@ void InMemoryStorage::CollectGarbage(std::unique_lock<utils::ResourceLock> main_
       main_lock_.lock_shared();
     }
   } else {
-    MG_ASSERT(main_guard.mutex() == std::addressof(main_lock_), "main_guard should be only for the main_lock_");
+    DMG_ASSERT(main_guard.mutex() == std::addressof(main_lock_), "main_guard should be only for the main_lock_");
```

Comment:
why DMG assert here

Reply:
Because it is dangerous to do these asserts in production and also slower, so this is a bit better solution as per @Ignition